### PR TITLE
In write_header() use poll if available.

### PR DESCRIPTION
**Pull request by deng02**
We are hitting the select bug where fileno > 1023
causing an "out of range" error.  Changing the select to a
poll where available to support higher numbered fileno.

### CODE REVIEW

File: clients/rospy/src/rospy/impl/tcpros_base.py [hunk 1]
```diff
@@ -613,11 +613,29 @@ def write_header(self):
...
+                for _, flag in events:
+                  if flag & select.POLLOUT:
+                        ready = True
+        else:
+            while not ready:
+                try:
+                    _, ready, _ = select.select([], [fileno], [])
+                except ValueError as e:
+                    logger.error("[%s]: select fileno '%s': %s", self.name, str(fileno), str(e))
+                    raise e
```
Comment (mikepurvis) [inline clients/rospy/src/rospy/impl/tcpros_base.py]: I believe you can just `raise` and it will automatically re-raise the current exception.

**Comment by mikepurvis** [comment]
Looks good to me. 👍

**Comment by deng02** [comment]
ping @tfoote @dirk-thomas

**Comment by mikepurvis** [comment]
Looks like this is a similar kind of change to what's going on in #833. @mgrrx, can you comment?

Am curious to better understand the relationship between this code and the `Poller` abstraction class seen here: https://github.com/ros/ros_comm/blob/674263142a0be1124b273bdb1e0a79c3c956e259/clients/rospy/src/rospy/topics.py#L185

**Comment by dirk-thomas** [comment]
Just a though: the code path if `poll` is not available likely needs to be tested in Windows.

**Pull request merged**